pinot-core/src/main/java/org/apache/pinot/core/plan/AggregationGroupByOrderByPlanNode.java

```diff
@@ -87,6 +101,11 @@ public AggregationGroupByOrderByPlanNode(IndexSegment indexSegment, QueryContext
 
     Set<ExpressionContext> expressionsToTransform =
         AggregationFunctionUtils.collectExpressionsToTransform(_aggregationFunctions, _groupByExpressions);
+    _enableGroupByOpt = checkOrderByOptimization();
```

Comment:
Seems most of the checks for enabling the optimization rely on the column metadata, and has to happen in the operator. Let's move the whole check inside then. Splitting the checks in 2 places is hard to manage